AutomatedTesting/Gem/CMakeLists.txt

```
@@ -6,6 +6,10 @@
 #
 #
 
+set(gem_path ${CMAKE_CURRENT_LIST_DIR})
+set(gem_json ${gem_path}/gem.json)
+o3de_restricted_path(${gem_json} gem_restricted_path gem_parent_relative_path)
```

Can you unset these variables afterwards or or directly invoke pass in ${CMAKE_CURRENT_LIST_DIR}?
CMake variable scoping extends to add_subdirectory calls, so there is a very high likely hood that this gem_path/gem_json variable will conflict with variables declared in child subdirectories.

```suggestion
o3de_restricted_path(${CMAKE_CURRENT_LIST_DIR}/gem.json gem_restricted_path gem_parent_relative_path)
```